flake.nix

```diff
@@ -1,144 +1,142 @@
 {
   description = "Common styles for NixOS related web sites.";
 
-  inputs.nixpkgs = { url = "nixpkgs/nixos-unstable"; };
+  inputs.nixpkgs = { url = "github:nixos/nixpkgs/nixos-21.11"; };
 
   outputs =
     { self
-    , nixpkgs 
+    , nixpkgs
     }:
     let
-      system = "x86_64-linux";
-      pkgs = import nixpkgs { inherit system; };
-      package = builtins.fromJSON (builtins.readFile ./package.json);
+      supportedSystems = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
 
-      flake = {
+      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; overlays = [ overlay ]; });
+      package = builtins.fromJSON (builtins.readFile ./package.json);
 
-        defaultPackage."${system}" = flake.packages."${system}".commonStyles;
+      overlay = final: prev: {
```

Comment:
If we don't expose overlay in flake we don't need to use it and we can just deal directly with the package.